transmogrifier/sources/xml/datacite.py

```diff
@@ -339,6 +238,189 @@ def _get_contributors(cls, source_record: Tag) -> Iterator[timdex.Contributor]:
                     kind=contributor.get("contributorType") or "Not specified",
                 )
 
+    @classmethod
+    def get_dates(
+        cls,
+        source_record: Tag,
+    ) -> list[timdex.Date] | None:
+        dates = []
+        dates.extend(list(cls._get_publication_year(source_record)))
+        dates.extend(list(cls._get_dates(source_record)))
+        return dates or None
+
+    @classmethod
+    def _get_publication_year(cls, source_record: Tag) -> Iterator[timdex.Date]:
+        if publication_year := source_record.metadata.find(
+            "publicationYear", string=True
+        ):
+            publication_year = str(publication_year.string.strip())
+            if validate_date(
+                publication_year,
+                cls.get_source_record_id(source_record),
+            ):
+                yield timdex.Date(kind="Publication date", value=publication_year)
+        else:
+            logger.warning(
+                "Datacite record %s missing required Datacite field publicationYear",
+                cls.get_source_record_id(source_record),
+            )
+
+    @classmethod
+    def _get_dates(cls, source_record: Tag) -> Iterator[timdex.Date]:
```

Comment:
Forgive me if we've already discussed this, but I keep tripping on this private method `_get_dates()` and it's relationship to `get_dates()`.  What about a rename to something like `_get_other_dates()`?  I realize that "other" isn't very exact, but I think it definitively differentiates from the higher level field  method `get_dates()`.

Reply:
What about get `_get_datacite_dates` which I think is more accurate? I could also update `_get_contributors` to `_get_datacite_contributors` since that's what originally prompted the conversation

Reply:
Ah thanks for the reminder, I thought it had come up.  

I don't mean to be difficult, but even adding `datacite` feels odd... like we know it's datacite already.   But I don't really have a better solution.  

I suppose one unique property is that we're setting the date `kind` dynamically by the attribute `dateType`; wonder if that could come into play naming wise?  Maybe something like `_get_dates_by_datetype_attribute()`?  It's _long_... but it tells you precisely what's going on?

Or we could focus on the element used, like `_get_dates_by_date_element()`?  

Open to pushback here.  If folks are okay with `_get_dates()` and `_get_contributors()` I could be too.

Reply:
I like `_get_dates_by_date_element`, I think I'll go with that